src/Engine/TranskribusClient.php

```diff
@@ -137,8 +149,89 @@ private function request(
                 return $content;
             }
             $this->throwException(0);
-        } else {
+        }
+
+        if (401 === $statusCode) {
+            $this->setAccessToken();
+
+            $body['headers']['Authorization'] = 'Bearer '.$this->accessToken;
+            $response = $this->httpClient->request($method, $url, $body);
+            $statusCode = $response->getStatusCode();
+
+            if (200 === $statusCode) {
+                $content = json_decode($response->getContent());
+
+                if (!empty($content)) {
+                    return $content;
+                }
+                $this->throwException(0);
```

Comment:
I know it's not being introduced in this patch, but these magic numbers are not great I think. Better to set the codes to be class constants and then use them here, e.g. `$this->throwException(self::ERROR_NO_CONTENT);` (can be done in a separate patch of course).